EngineLayer/FdrAnalysis/PEPValueAnalysisGeneric.cs

```diff
@@ -135,63 +146,150 @@ public static string ComputePEPValuesForAllPSMsGeneric(List<PeptideSpectralMatch
             //mlContext.Model.Save(trainedModel, trainingData.Schema, @"C:\Users\User\Downloads\TrainedModel.zip");
         }
 
-        private static Dictionary<string, int> GetSequenceToPSMCount(List<PeptideSpectralMatch> psms)
+        /// <summary>
+        /// Assuming that we want to use different sets of training features for different search types, the search type needs to be determined.
+        /// In the future, we may have that specified in the GUI and this will be moot.
+        /// </summary>
+        /// <param name="psms"></param>
+        /// <returns></returns>
+        private static string DetermineSearchType(List<PeptideSpectralMatch> psms)
         {
-            Dictionary<string, int> sequenceToPsmCount = new Dictionary<string, int>();
-
-            List<string> sequences = new List<string>();
-            foreach (PeptideSpectralMatch psm in psms)
+            if (psms[1].DigestionParams.Protease.Name == "top-down")
             {
-                var ss = psm.BestMatchingPeptides.Select(b => b.Peptide.FullSequence).ToList();
-                sequences.Add(String.Join("|", ss));
+                return "topDown";
             }
-
-            var s = sequences.GroupBy(i => i);
-
-            foreach (var grp in s)
+            else
             {
-                sequenceToPsmCount.Add(grp.Key, grp.Count());
+                return "standard";
             }
-            return sequenceToPsmCount;
         }
 
         /// <summary>
-        /// This method goes throught the complete set of peptide spectral matches and counts the number of appearances for each accession number.
-        /// This includes all of the ambiguous entries. So, if one peptide spectral match has three peptides with three accessions, all are counted.
+        /// This method can be used to select a custom subset of features for use in the Microsoft ML.NET computation of posterior error probability
+        /// Probably should have unique sets for regular, crosslinking, top-down, labelled.
         /// </summary>
-        /// <param name="psms"></param>
         /// <returns></returns>
-        public static Dictionary<string, int> GetAccessionCounts(List<PeptideSpectralMatch> psms)
+        private static string[] GetCustomFeatures(string searchType)
         {
-            Dictionary<string, int> accessionCountDictionary = new Dictionary<string, int>();
+            //non-specific and top-down searches don't used missedCleavages
+            switch (searchType)
+            {
+                case "topDown":
+                    return new string[] { "Intensity", "ScanPrecursorCharge", "DeltaScore", "Notch", "PsmCount", "ModsCount", "MissedCleavagesCount", "Ambiguity", "LongestFragmentIonSeries" };
```

Comment:
Comment above states that top-down searches don't use missedCleavages, but MissedCleavageCount is still included here. I was wondering why this is the case.

Reply:
deleted missed cleavages